plugins/uefi-dbx/fu-dbxtool.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupd.h>
+#include <glib/gi18n.h>
+#include <glib-unix.h>
+#include <locale.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "fu-efivar.h"
+#include "fu-uefi-dbx-common.h"
+#include "fu-uefi-dbx-file.h"
+
+/* custom return code */
+#define EXIT_NOTHING_TO_DO		2
+
+static void
+fu_util_ignore_cb (const gchar *log_domain, GLogLevelFlags log_level,
+		   const gchar *message, gpointer user_data)
+{
+}
+
+int
+main (int argc, char *argv[])
+{
+	GPtrArray *checksums_system;
+	GPtrArray *checksums_update;
+	gboolean action_apply = FALSE;
+	gboolean action_list = FALSE;
+	gboolean force = FALSE;
+	gboolean verbose = FALSE;
+	gsize bufsz = 0;
+	g_autofree gchar *dbxfile = NULL;
+	g_autofree guint8 *buf_system = NULL;
+	g_autofree guint8 *buf_update = NULL;
+	g_autoptr(FuUefiDbxFile) dbx_system = NULL;
+	g_autoptr(FuUefiDbxFile) dbx_update = NULL;
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GOptionContext) context = NULL;
+	const GOptionEntry options[] = {
+		{ "verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose,
+			/* TRANSLATORS: command line option */
+			_("Show extra debugging information"), NULL },
+		{ "list", 'l', 0, G_OPTION_ARG_NONE, &action_list,
+			/* TRANSLATORS: command line option */
+			_("List entries in dbx"), NULL },
+		{ "apply", 'a', 0, G_OPTION_ARG_NONE, &action_apply,
+			/* TRANSLATORS: command line option */
+			_("Apply update files"), NULL },
+		{ "dbx", 'd', 0, G_OPTION_ARG_STRING, &dbxfile,
+			/* TRANSLATORS: command line option */
+			_("Specify the dbx database file"), "FILENAME" },
+		{ "force", 'f', 0, G_OPTION_ARG_NONE, &force,
+			/* TRANSLATORS: command line option */
+			_("Apply update even when not advised"), NULL },
+		{ NULL}
+	};
+
+	setlocale (LC_ALL, "");
+
+	bindtextdomain (GETTEXT_PACKAGE, FWUPD_LOCALEDIR);
+	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+	textdomain (GETTEXT_PACKAGE);
+
+	/* ensure root user */
+#ifdef HAVE_GETUID
+	if (getuid () != 0 || geteuid () != 0)
+		/* TRANSLATORS: we're poking around as a power user */
+		g_printerr ("%s\n", _("This program may only work correctly as root"));
```

Comment:
Since this app only support apply and list, I think that this check should only be shown with the apply method, right?

Reply:
Fixed.